swarm/storage/mru/signedupdate.go

```diff
@@ -48,26 +53,24 @@ func (r *SignedResourceUpdate) Verify() (err error) {
 	}
 
 	// get the address of the signer (which also checks that it's a valid signature)
-	ownerAddr, err := getOwner(digest, *r.signature)
+	r.viewID.ownerAddr, err = getOwner(digest, *r.signature)
 	if err != nil {
 		return err
 	}
 
+	// check that the lookup information contained in the chunk matches the updateAddr (chunk search key)
+	// that was used to retrieve this chunk
+	// if this validation fails, someone forged a chunk.
 	if !bytes.Equal(r.updateAddr, r.UpdateAddr()) {
-		return NewError(ErrInvalidSignature, "Signature address does not match with ownerAddr")
-	}
-
-	// Check if who signed the resource update really owns the resource
-	if !verifyOwner(ownerAddr, r.metaHash, r.rootAddr) {
-		return NewErrorf(ErrUnauthorized, "signature is valid but signer does not own the resource: %v", err)
+		return NewError(ErrInvalidSignature, "Signature address does not match with update ownerAddr")
```

Comment:
I think I asked about this before. My apologies if that's the case;

This is to protect the node's reputation from naughty clients trying to spoof? (Invalid chunks will be stopped by the peers).